### PR TITLE
Timestamp

### PR DESCRIPTION
**Pull request by Obinnabii**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/PostCommentServlet.java [hunk 1]
```diff
@@ -39,12 +39,13 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String text = request.getParameter("comment-input");
     if (!text.isBlank()) { 
+      text = text.trim();
       long commentPostTime = System.currentTimeMillis();
       Entity commentEntity = new Entity(Comment.ENTITY_NAME);
-
+
```
Comment (billyeh) [inline portfolio/src/main/java/com/google/sps/servlets/PostCommentServlet.java]:
```suggestion

```

File: portfolio/src/main/java/com/google/sps/servlets/PostCommentServlet.java [hunk 2]
```diff
@@ -39,12 +39,13 @@
...
     if (!text.isBlank()) { 
+      text = text.trim();
       long commentPostTime = System.currentTimeMillis();
       Entity commentEntity = new Entity(Comment.ENTITY_NAME);
-
+      
       commentEntity.setProperty(Comment.TEXT_FIELD, text);
       commentEntity.setProperty(Comment.POST_TIME_FIELD, commentPostTime);
-
+
```
Comment (billyeh) [inline portfolio/src/main/java/com/google/sps/servlets/PostCommentServlet.java]:
```suggestion

```

File: portfolio/src/main/java/com/google/sps/servlets/PostCommentServlet.java [hunk 3]
```diff
@@ -39,12 +39,13 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String text = request.getParameter("comment-input");
     if (!text.isBlank()) { 
+      text = text.trim();
```
Comment (billyeh) [inline portfolio/src/main/java/com/google/sps/servlets/PostCommentServlet.java]:
```suggestion
    String text = request.getParameter("comment-input").trim();
    if (!text.isBlank()) { 
```
Maybe? If we don't want to post empty comments

File: portfolio/src/main/webapp/comments.js [hunk 4]
```diff
@@ -93,7 +96,22 @@ function createCommentElement(text) {
...
-/** Posts a comment to the server */
+/** 
+ * Create the container for the timestamp of individual posts
+ * @param {int} postTime in milliseconds
+ * @return {HTMLElement} the html element containing the @param postTime
+ */
+function createPostTimeElement(postTime) {
+  let postTimeElement = document.createElement('small');
+  let postTimeString = '\n' + new Date(postTime).toLocaleString();
+  postTimeElement.innerText = postTimeString
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]:
```suggestion
  postTimeElement.innerText = postTimeString;
```

File: portfolio/src/main/webapp/comments.js [hunk 5]
```diff
@@ -93,7 +96,22 @@ function createCommentElement(text) {
...
 
-/** Posts a comment to the server */
+/** 
+ * Create the container for the timestamp of individual posts
+ * @param {int} postTime in milliseconds
+ * @return {HTMLElement} the html element containing the @param postTime
+ */
+function createPostTimeElement(postTime) {
+  let postTimeElement = document.createElement('small');
+  let postTimeString = '\n' + new Date(postTime).toLocaleString();
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js:106]: https://stackoverflow.com/questions/13274279/authentication-filter-and-servlet-for-login/13276996#13276996

Comment (billyeh) [inline portfolio/src/main/webapp/comments.js:107]: Comment explaining why '\n' is needed here would be good

**Review by billyeh — APPROVED** [review]

File: portfolio/src/main/webapp/comments.js (hunk 5 above)
Comment (Obinnabii) [reply portfolio/src/main/webapp/comments.js:107]:
```suggestion
  // Insert a newline character to separate the comment from its timestamp
  let postTimeString = '\n' + new Date(postTime).toLocaleString();
```

**Pull request merged**